packages/lesswrong/components/recentDiscussion/RecentDiscussionFeed.tsx

```diff
@@ -92,6 +93,14 @@ const RecentDiscussionFeed = ({
   const refetchRef = useRef<null|ObservableQuery['refetch']>(null);
   const currentUser = useCurrentUser();
   const expandAll = currentUser?.noCollapseCommentsFrontpage || expandAllThreads
+  
+  const [sessionId] = useState<string>(() => {
+    if (typeof window === 'undefined') return randomId();
+    const storage = window.sessionStorage;
+    const currentId = storage ? storage.getItem('recentDiscussionSessionId') ?? randomId() : randomId();
+    storage.setItem('recentDiscussionSessionId', currentId);
+    return currentId;
+  });
```

Comment:
I'm confused by this block:
1. why `useState` if you're not using a setter?
2. Do we even render `RecentDiscussionFeed` on the server?
3. why `sessionStorage` instead of local storage?  Is it because we actively prefer to not persist session ids beyond the current tab lifetime?  (If so, legit)

Reply:
1. you're right, should use something else
2. yes, it is SSR'd on LessWrong
3. that's right, it's per tab not per device. this is the same logic as is for UltraFeed